internal/bed/sorting.go

```diff
@@ -88,3 +109,42 @@ func naturalStringCompare(a, b string) int {
 	}
 	return 1
 }
+
+// Compares string using a predefined order contained in a slice.
+//
+// Strings that are not in the slice with be ranked as greater
+// than the strings in the slice. If neither a or b are in the slice
+// they will be compared using naturalStringCompare.
+//
+//	-1 if a is less than b
+//	 0 if a equals b
+//	+1 if a is greater than b
+func stringSliceCompare(a, b string, order []string) int {
+	a = strings.ToLower(a)
+	b = strings.ToLower(b)
+
+	aIdx := idxInSlice(order, a)
+	bIdx := idxInSlice(order, b)
+
+	if aIdx != -1 && bIdx != -1 {
+		return cmp.Compare(aIdx, bIdx)
+	}
+	if aIdx != -1 && bIdx == -1 {
+		return -1
+	}
+	if bIdx != -1 && aIdx == -1 {
+		return 1
+	}
+	return naturalStringCompare(a, b)
+}
+
+// Returns position if item is in slice and
+// -1 if item is not in slice
+func idxInSlice(slice []string, item string) int {
```

Comment:
If you made the slice a map containing the names as keys and the indices as values, you don't need to do the ranging. That would probably increase speed when sorting long bed files. You need to be aware that missing keys will return zero (default value for int) so maybe start the indices at 1. Another way would be to check if the key exists and simply handle the case where it doesn't.

Reply:
This made me think a bit. 
What if we let the user provide a custom chromosome ordering as a list and then we transform it into a map for sorting? We can use the human sorting as default and then people working with non-human organisms, can also get the chromosome ordering they want?

Reply:
I think that is actually a great idea! 👍